stonesoup/dataassociator/_assignment.py

```diff
@@ -0,0 +1,319 @@
+    # Tracker Component Library of
+    # https://github.com/USNavalResearchLaboratory/TrackerComponentLibrary
+    #
+    # EXAMPLE 1:
+    # import numpy
+    # import assignAlgs
+    # Inf=numpy.inf
+    # C=numpy.array([[Inf,   2, Inf, Inf, 3],
+    #              [   7, Inf,  23, Inf, Inf],
+    #              [  17,  24, Inf, Inf, Inf],
+    #              [ Inf,   6,  13,  20, Inf]])
+    # maximize=False
+    # gain, col4row, row4col=assignAlgs.assign2D(C,maximize)
+    # print(gain)
+    # One will get an optimal assignment having a gain of 47
+    #
+    # EXAMPLE 2:
+    # This is the example used in [3]. Here, we demonstrate how to form
+    # assignment tuples (index from 1, not 0) from col4row.
+    # import numpy
+    # import assignAlgs
+    # C=numpy.array([[7,   51,  52,  87,  38,  60,  74,  66,   0,  20],
+    #               [50,  12,   0,  64,   8,  53,   0,  46,  76,  42],
+    #               [27,  77,   0,  18,  22,  48,  44,  13,   0,  57],
+    #               [62,   0,   3,   8,   5,   6,  14,   0,  26,  39],
+    #               [0,   97,   0,   5,  13,   0,  41,  31,  62,  48],
+    #               [79,  68,   0,   0,  15,  12,  17,  47,  35,  43],
+    #               [76,  99,  48,  27,  34,   0,   0,   0,  28,   0],
+    #               [0,   20,   9,  27,  46,  15,  84,  19,   3,  24],
+    #               [56,  10,  45,  39,   0,  93,  67,  79,  19,  38],
+    #               [27,   0,  39,  53,  46,  24,  69,  46,  23,   1]])
+    # maximize=False
+    # gain, col4row, row4col=assignAlgs.assign2D(C,maximize)
+    # tuples=numpy.empty((2,10),dtype=int)
+    # for curRow in range(0,10):
+    #    tuples[0,curRow]=curRow+1
+    #    tuples[1,curRow]=col4row[curRow]+1
+    # print(gain)
+    # print(tuples)
+    # One will see that the gain is 0 and the assigned  tuples match what
+    # is in [3]. However, the assigned tuples is NOT obtained by attaching
+    # col4row to row4col.
+    #
+    # REFERENCES:
+    # [1]   D. F. Crouse, "On Implementing 2D Rectangular Assignment
+    #       Algorithms," IEEE Transactions on Aerospace and Electronic
+    #       Systems, vol. 52, no. 4, pp. 1679-1696, Aug. 2016.
+    # [2]   D. F. Crouse, "Advances in displaying uncertain estimates of
+    #       multiple targets," in Proceedings of SPIE: Signal Processing,
+    #       Sensor Fusion, and Target Recognition XXII, vol. 8745,
+    #       Baltimore, MD, Apr. 2013.
+    # [3]   Murty, K. G. "An algorithm for ranking all the assignments in
+    #       order of increasing cost," Operations Research, vol. 16, no. 3,
+    #       pp. 682-687, May-Jun. 1968.
+    #
+    # May 2018 David F. Crouse, Naval Research Laboratory, Washington D.C.
+    # (UNCLASSIFIED) DISTRIBUTION STATEMENT A. Approved for public release.
+    # This work was supported by the Office of Naval Research through the
+    # Naval Research Laboratory 6.1 Base Program
+
+    C = numpy.matrix(C)  # In case a multidimensional array was passed.
```

Comment:
This seems to throw a numpy PendingDeprecationWarning:

```
PendingDeprecationWarning: the matrix subclass is not the recommended way to represent matrices or deal with linear algebra (see https://docs.scipy.org/doc/numpy/user/numpy-for-matlab-users.html). Please adjust your code to use regular ndarray.
    C = numpy.matrix(C)  # In case a multidimensional array was passed.
```

Might be worth fixing